### ISSUE TITLE
Script can't apt install [jail DNS issue]

### ISSUE THREAD

**Issue by stayprivate** [saveriomiroddi/zfs-installer#163]
Hello,

Running the script with Ubuntu server 20.  When Ubuntu is done with it's installation, I got back to the zfs-installer, press enter to resume the procedure. It gets to install_jail_zfs_backages_UbuntuServer, runs what look likes apt install and then complain that it can't reach the site.
  Err:1 http:://ca.archive.ubuntu.com/ubuntu focal blabla, 
   Temporary failure resolving ca.archive.ubuntu.com
 ..

The link is that of the mirror site specify during the ubuntu install.   What I don't understand is that in another console (CTRL-ALT-F3) I run an apt install libnvpair1linux, it works just fine. The Ubuntu installer had no issue downloading/installing all updates.  For some reason the zfs-installer can't.   I try to manual do a apt update before resuming the zfs-install, to now avail.   I try to specify a different mirror, no change. 

More puzzling to me. I kill the installer, in the same shell do a sudo apt install someting, which work. I hack the script to get it to bypass all of the setup to get to install_jail_zfs_packages.. At startup it does an apt install that works.  then when it gets to install_jail_zfs_packages it fails, in the same way!!!

Suggestions?

**Comment by saveriomiroddi** [comment]
Hello!

At that stage, the installer runs commands inside the jail, which has its own configuration (specifically, it uses a manual DNS server configuration, Google's `8.8.8.8`). This is very standard procedure; I wonder if a recent update introduced some shenanigan that messes with the nameserver.

You can verify this by running, after the failure, `sudo chroot /mnt "ping -c 1 8.8.8.8"`, which should succeed, and `sudo chroot /mnt "ping -c www.google.com"`, which should fail.

I can't verify this now; I'll have a look over the next week.

If you want to try to figure out why this is not working, the subject to search is "ubuntu chroot nameserver" -

**Comment by stayprivate** [comment]
That was fast , much appreciated.

Interesting, turns out IT here is blocking 8.8.8.8.

I modified the script to use our internal dns and it kind of work. I also had to add a chroot_execute "apt update" in install_jail_zfs_packages_UbuntuServer as it would not really recognized the mirror site. I now have a VM with Ubuntu 20.04 and zfs, oh yeah !  Next step try to customize the ubuntu server with Cubic to have this done automagicaly. 

Thanks !
